nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/hadoop/SecurityUtil.java

```diff
@@ -151,7 +151,19 @@ public static boolean isSecurityEnabled(final Configuration config) {
 
     public static <T> T callWithUgi(UserGroupInformation ugi, PrivilegedExceptionAction<T> action) throws IOException {
         try {
-            return ugi.doAs(action);
+            T result;
+            if (ugi == null) {
+                try {
+                    result = action.run();
+                } catch (RuntimeException re) {
+                    throw re;
+                } catch (Exception e) {
```

Comment:
I think `IOException` should be handled separately, thrown as-is without wrapping it in a `RuntimeException`. Adding `IOException` to the previous `catch` statement would be enough.